nvtabular/workflow/node.py

```diff
@@ -133,54 +224,33 @@ def __add__(self, other):
         -------
         WorkflowNode
         """
-        left_arg = self
-        right_arg = other
-
-        added_node = None
-        added_selector = None
-
-        if isinstance(right_arg, WorkflowNode):
-            # If an argument is already an addition node, make it
-            # the left arg and combine the right arg into it
-            if isinstance(other.op, internal.ConcatColumns):
-                left_arg = other
-                right_arg = self
-
-            added_node = right_arg
-            added_selector = right_arg.output_columns
-        elif isinstance(right_arg, ColumnSelector):
-            added_selector = right_arg
-        elif isinstance(right_arg, list):
-            added_selector = ColumnSelector()
-            for element in right_arg:
-                if isinstance(element, WorkflowNode):
-                    added_selector += element.output_columns
-                else:
-                    added_selector += element
-            added_selector = ColumnSelector(subgroups=added_selector)
-        else:
-            added_selector = ColumnSelector(right_arg)
-
-        if isinstance(left_arg.op, internal.ConcatColumns):
-            child = left_arg
+        if isinstance(self.op, internal.ConcatColumns):
+            child = self
         else:
-            child = WorkflowNode(left_arg.output_columns)
+            # Create a child node
+            child = WorkflowNode()
             child.op = internal.ConcatColumns(label="+")
 
-            left_arg.children.append(child)
-            child.parents.append(left_arg)
-
-        child.selector += added_selector
-
-        if added_node:
-            if isinstance(added_node.op, internal.ConcatColumns):
-                child.parents += added_node.parents
-                for parent in added_node.parents:
-                    parent.children.append(child)
-                    parent.children.remove(added_node)
-            else:
-                child.parents.append(added_node)
-                added_node.children.append(child)
+            # Add self as a parent
+            self.children.append(child)
+            child.parents.append(self)
+
+        # The right operand becomes a dependency
+        if isinstance(other, list):
+            converted_elem = []
+            for elem in other:
+                if not isinstance(elem, (ColumnSelector, WorkflowNode)):
+                    elem = ColumnSelector(elem)
```

Comment:
This is another frequently repeated piece of code that iterates through and converts strings to selectors. Maybe another extractable method?